auslib/web/public/base.py

```diff
@@ -61,10 +61,12 @@ def apply_security_headers(response):
 
 @app.errorhandler(404)
 def fourohfour(error):
-    """We don't return 404s in AUS. Instead, we return empty XML files"""
-    response = make_response('<?xml version="1.0"?>\n<updates>\n</updates>')
-    response.mimetype = 'text/xml'
-    return response
+    if re.match("^/update", request.path):
+        """We don't return 404s in AUS. Instead, we return empty XML files"""
+        response = make_response('<?xml version="1.0"?>\n<updates>\n</updates>')
+        response.mimetype = 'text/xml'
+        return response
+    return Response(status=404, mimetype="text/plain")
```

Comment:
I think it would be better to return the original error instead of making your own - you may be losing some information by doing this. You might need to raise it instead of returning, I don't recall exactly how this works...